src/ctia/stores/es/crud.clj

```diff
@@ -306,32 +311,107 @@ It returns the documents with full hits meta data including the real index in wh
                                            access-control-filter-list
                                            ident))))))
 
+
+(s/defn make-search-query
+  [{{:keys [default_operator]} :props} :- ESConnState
+   {:keys [query-string filter-map date-range]} :- SearchQuery
+   ident]
+  (let [es-query-string {:query_string (into {:query query-string}
+                                             (when default_operator
+                                               {:default_operator default_operator}))}
+        date-range-query (when date-range
+                           {:range date-range})
+        filter-terms (-> (ensure-document-id-in-map filter-map)
+                         q/prepare-terms)]
+    {:bool
+     {:filter
+      (cond-> [(find-restriction-query-part ident)]
+        (seq filter-map) (into filter-terms)
+        (seq date-range) (conj date-range-query)
+        (not-empty query-string) (conj es-query-string))}}))
```

Comment:
why `not-empty`?

generally I use `seq` as a predicate, and `not-empty` only for obtaining an object I intend to further use

The following seems to apply:

```
user=> (doc empty?)
-------------------------
clojure.core/empty?
([coll])
  Returns true if coll has no items - same as (not (seq coll)).
  Please use the idiom (seq x) rather than (not (empty? x))
```